src/memory-analyzer/analyze_symbol.h

```diff
@@ -36,64 +39,141 @@ class gdb_apit;
 class exprt;
 class source_locationt;
 
+/// Interface for extracting values from GDB (building on \ref gdb_apit)
 class symbol_analyzert
 {
 public:
   symbol_analyzert(const symbol_tablet &symbol_table, gdb_apit &gdb_api);
 
+  /// For each input symbol in \p symbols: map its value address to its
+  ///   \ref symbol_exprt (via the `values` map) and then call
+  ///   \ref analyze_symbol on it.
+  /// \param symbols: names of symbols to be analysed
   void analyze_symbols(const std::vector<std::string> &symbols);
 
+  /// Get memory snapshot as C code
+  /// \return converted block of code with the collected assignments
   std::string get_snapshot_as_c_code();
+
+  /// Get memory snapshot as symbol table
+  /// Build a new \ref symbol_tablet and for each `lhs` symbol from collected
+  ///   assignments [lhs:=rhs] store a new symbol (with the `rhs` as value)
+  ///   there. Also, type symbols are copied from \ref symbol_table.
+  /// \return a new symbol table with known symbols having their extracted
+  ///   values
   symbol_tablet get_snapshot_as_symbol_table();
 
   using pointer_valuet = gdb_apit::pointer_valuet;
 
 private:
   gdb_apit &gdb_api;
+
+  /// External symbol table -- extracted from \ref read_goto_binary
+  /// We only expect to analyse symbols located there.
   symbol_tablet symbol_table;
   const namespacet ns;
   expr2ct c_converter;
   allocate_objectst allocate_objects;
 
+  /// Sequence of assignments collected during \ref analyze_symbols
   std::vector<std::pair<exprt, exprt>> assignments;
 
+  /// Mapping memory location (from \ref gdb_apit) to pointer expression for
+  ///   which \ref get_non_char_pointer_value returned nil expression.
   std::map<exprt, std::string> outstanding_assignments;
+
+  /// Storing pairs <address, symbol> such that at `address` is stored the
+  ///   value of `symbol`.
   std::map<std::string, exprt> values;
 
+  /// Assign the gdb-extracted value for \p symbol_name to its symbol
+  ///   expression and then process outstanding assignments that this
+  ///   extraction introduced.
+  /// \param symbol_name: symbol table name to be analysed
   void analyze_symbol(const std::string &symbol_name);
 
+  /// Create assignment \p lhs := \p value (see \ref analyze_symbol)
+  /// \param lhs: the left-hand side of the assignment; expected to be a
+  ///             \ref symbol_exprt
+  /// \param value: the value to be assigned; the result of \ref get_expr_value
   void add_assignment(const exprt &lhs, const exprt &value);
 
+  /// Iterate over \p array and fill its operands with the results of calling
+  ///   \ref get_expr_value on index expressions into \p expr.
+  /// \param expr: the expression to be analysed
+  /// \param array: array with zero-initialised elements
+  /// \param location: not used
```

Comment:
I think we should here just say "the source location". The location is not directly used but it's nevertheless passed on to callees. (and same for other cases in this PR)

Reply:
Done.